src/EventStore.Core/GrpcConnectionListener.cs

```diff
@@ -0,0 +1,72 @@
+using System.Collections.Generic;
+using System.Diagnostics.Metrics;
+using System.Diagnostics.Tracing;
+
+namespace EventStore.Core;
+
+public class GrpcConnectionListener : EventListener {
+	private readonly Dictionary<string, bool> _connections = new();
+	private readonly UpDownCounter<long> _grpcConnectionsMetric;
+
+	public GrpcConnectionListener(UpDownCounter<long> grpcConnectionsMetric) {
```

Comment:
much to my surprise, https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.tracing.eventlistener?view=net-7.0 says 

> When instantiating an EventListener, the callbacks to [OnEventSourceCreated](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.tracing.eventlistener.oneventsourcecreated?view=net-7.0) and [OnEventWritten](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.tracing.eventlistener.oneventwritten?view=net-7.0) can happen before the constructor has completed. Take care when you initialize instance members used in those callbacks.